libs/npm/core/src/components/grid-table/GridTable.tsx

```diff
@@ -312,23 +338,41 @@ export const GridTable = <T extends object>({
         ))}
       </div>
       <div {...getTableBodyProps()}>
-        {page.map((row) => {
+        {page.map((row, index) => {
           prepareRow(row);
-          return (
-            <div
-              {...row.getRowProps()}
-              onClick={(e) => handleRowClick(e, row)}
-              className={onRowRenderClassName(row)}
-            >
-              {row.cells.map((cell) => {
-                return (
-                  <div {...cell.getCellProps()}>
-                    {cell.render('Cell') as unknown as React.ReactNode}
-                  </div>
-                );
-              })}
-            </div>
-          );
+          if (page.length === index + 1 && infiniteScroll) {
```

Comment:
Is there a way to write the two outputs?  They are identical except the `ref={lastRowRef}`.  It makes it difficult to know what's really different and then will introduce maintenance issues down the road.